### PR TITLE
PIM-3851: List Reference Data to be able to link them to a Product Value

### PR DESCRIPTION
**Pull request by jjanvier**
| Q                    | A
| -------------------- | ---
| Bug fix?             |
| New feature?         |
| BC breaks?           |
| CI currently passes? |
| Tests pass?          | Y
| Scenarios pass?      |
| Checkstyle issues?*  |
| PMD issues?**        |
| Changelog updated?   |
| Fixed tickets        |
| DB schema updated?   |
| Migration script?    |
| Doc PR               |

### CODE REVIEW

File: src/Pim/Bundle/UIBundle/Controller/AjaxOptionController.php [hunk 1]
```diff
@@ -40,13 +39,28 @@ public function __construct(RegistryInterface $doctrine)
...
+            $choices = $repository->getOptions(
                 $query->get('dataLocale'),
                 $query->get('collectionId'),
                 $query->get('search'),
                 $query->get('options', array())
             );
+        } elseif ($repository instanceof ReferenceDataRepositoryInterface) {
+            $choices['results'] = $repository->findBySearch(
+                $query->get('search'),
+                $query->get('options', array())
```
Comment (skeleton) [inline src/Pim/Bundle/UIBundle/Controller/AjaxOptionController.php]: use []

File: src/Pim/Bundle/UIBundle/Form/Transformer/AjaxReferenceDataTransformer.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+namespace Pim\Bundle\UIBundle\Form\Transformer;
+
+use Pim\Component\ReferenceData\Repository\ReferenceDataRepositoryInterface;
+use Symfony\Component\Form\DataTransformerInterface;
+
+/**
+ * Ajax choice transformer for reference data
+ *
+ * @author    Julien Janvuer <jjanvier@akeneo.com>
```
Comment (skeleton) [inline src/Pim/Bundle/UIBundle/Form/Transformer/AjaxReferenceDataTransformer.php]: Janvuer ;)

File: src/Pim/Bundle/UIBundle/Form/Transformer/AjaxReferenceDataTransformer.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+     * {@inheritdoc}
+     */
+    public function reverseTransform($value)
+    {
+        if ($this->options['multiple']) {
+            if (!$value) {
+                return array();
+            }
+
+            $values = array();
```
Comment (skeleton) [inline src/Pim/Bundle/UIBundle/Form/Transformer/AjaxReferenceDataTransformer.php]: use []

File: src/Pim/Bundle/UIBundle/Form/Transformer/AjaxReferenceDataTransformer.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            foreach (explode(',', $value) as $id) {
+                $values[] = $this->repository->findOneBy(['id' => $id]);
```
Comment (skeleton) [inline src/Pim/Bundle/UIBundle/Form/Transformer/AjaxReferenceDataTransformer.php]: Why do not use find() method ?

Comment (jjanvier) [inline src/Pim/Bundle/UIBundle/Form/Transformer/AjaxReferenceDataTransformer.php]: @skeleton because I'm stupid ! thanks

**Pull request merged**